cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceProgramRunner.java

```diff
@@ -224,6 +226,8 @@ public ProgramController run(final Program program, ProgramOptions options) {
                                                         final Arguments arguments, final Arguments userArgs) {
 
     final String twillRunId = arguments.getOption(ProgramOptionConstants.TWILL_RUN_ID);
+    final RetryStrategy retryStrategy = SystemArguments.getRetryStrategy(userArgs.asMap(), ProgramType.MAPREDUCE,
```

Comment:
BasicMapReduceContext should have calculated this already, we can get it from the context and pass it to this method.